packit_service/config.py

```diff
@@ -0,0 +1,167 @@
+
+        self.deployment: Deployment = Deployment.stg
+
+        self.github_app_id: Optional[str] = None
+        self.github_app_cert_path: Optional[str] = None
+        self._github_token: str = ""
+        self.webhook_secret: str = ""
+        self.validate_webhooks: bool = True
+
+        # %%% ACTIONS HANDLER CONFIGURATION %%%
+        # these values are specific to packit service when we run actions in a sandbox
+
+        # name of the handler to run actions and commands, default to current env
+        self.command_handler: RunCommandType = RunCommandType.local
+        # a dir where the PV is mounted: both in sandbox and in worker
+        self.command_handler_work_dir: str = ""
+        # name of the PVC so that the sandbox has the same volume mounted
+        self.command_handler_pvc_env_var: str = ""  # pointer to pointer, lol
+        # name of sandbox container image
+        self.command_handler_image_reference: str = SANDCASTLE_IMAGE
+        # do I really need to explain this?
+        self.command_handler_k8s_namespace: str = SANDCASTLE_DEFAULT_PROJECT
+
+        # path to a file where OGR should store HTTP requests
+        self.github_requests_log_path: str = ""
+
+    @classmethod
+    def get_from_dict(cls, raw_dict: dict, validate=True) -> "Config":
+        if validate:
+            cls.validate(raw_dict)
+
+        config = Config()
+
+        config.debug = raw_dict.get("debug", False)
+        config.dry_run = raw_dict.get("dry_run", False)
+        config.fas_user = raw_dict.get("fas_user", None)
+        config.keytab_path = raw_dict.get("keytab_path", None)
+        config._pagure_user_token = raw_dict.get("pagure_user_token", "")
+        config._pagure_fork_token = raw_dict.get("pagure_fork_token", "")
+        config.github_app_id = raw_dict.get("github_app_id", "")
+        config.github_app_cert_path = raw_dict.get("github_app_cert_path", "")
+        config.webhook_secret = raw_dict.get("webhook_secret", "")
+        config.deployment = Deployment.from_str(raw_dict.get("deployment", ""))
+        config.validate_webhooks = raw_dict.get("validate_webhooks", False)
+
+        config.command_handler = RunCommandType.local
+        a_h = raw_dict.get("command_handler")
+        if a_h:
+            config.command_handler = RunCommandType(a_h)
+        config.command_handler_work_dir = raw_dict.get(
+            "command_handler_work_dir", SANDCASTLE_WORK_DIR
+        )
+        config.command_handler_pvc_env_var = raw_dict.get(
+            "command_handler_pvc_env_var", SANDCASTLE_PVC
+        )
+        config.command_handler_image_reference = raw_dict.get(
+            "command_handler_image_reference", SANDCASTLE_IMAGE
+        )
+        # default project for oc cluster up
+        config.command_handler_k8s_namespace = raw_dict.get(
+            "command_handler_k8s_namespace", SANDCASTLE_DEFAULT_PROJECT
+        )
+
+        return config
+
+    @classmethod
+    def get_service_config(cls) -> "Config":
```

Comment:
I'd say that having this method here is an overkill: I'd say it's just enough to have a default path and make it changeable via an env var.